lib/travis/build/script/dart.rb

```diff
@@ -4,13 +4,53 @@ class Script
       class Dart < Script
         DEFAULTS = {
           dart: 'stable',
-          with_content_shell: 'false'
+          with_content_shell: false,
+          install_dartium: false,
+          xvfb: true
         }
 
+        attr_reader :task
+
+        def initialize(*args)
+          super
+
+          @task = config[:dart_task] || {}
+          @task = {task.to_sym => true} if task.is_a?(String)
+          @task[:install_dartium] = config[:install_dartium] unless task.include?(:install_dartium)
+          @task[:xvfb] = config[:xvfb] unless task.include?(:xvfb)
+          @task[:dart] ||= config[:dart]
+
+          # Run "pub run test" by default if no other tasks are specified.
+          @task[:test] ||= true if !@task[:dartanalyzer] && !@task[:dartfmt]
+        end
+
         def configure
           super
 
-          if with_content_shell
+          if config[:with_content_shell]
+            if config.include?(:dart_task)
+              sh.failure "with_content_shell can't be used with dart_task."
```

Comment:
Do keep in mind, though, that `sh.terminate` will also terminate the build immediately, without other phases (say, `after_failure`, `after_script`) running.